packages/adp-tooling/src/writer/changes/writers/data-source-writer.ts

```diff
@@ -56,28 +56,32 @@ export class DataSourceWriter implements IWriter<DataSourceData> {
      * @returns {Promise<void>} A promise that resolves when the change writing process is completed.
      */
     async write(data: DataSourceData): Promise<void> {
-        const { dataSourcesDictionary, service } = data;
-        const content = this.constructContent(service.name, service.uri, service.maxAge);
-        const change = getGenericChange(data, content, ChangeType.CHANGE_DATA_SOURCE);
+        const { variant, dataSources, answers } = data;
+        const { id, uri, maxAge, annotationUri } = answers;
+        const annotationId = dataSources[id].settings?.annotations?.[0];
+
+        const timestamp = Date.now();
+        const content = this.constructContent(id, uri, maxAge);
+        const change = getChange(variant, timestamp, content, ChangeType.CHANGE_DATA_SOURCE);
 
         writeChangeToFolder(
             this.projectPath,
             change,
-            `id_${data.timestamp}_changeDataSource.change`,
+            `id_${timestamp}_changeDataSource.change`,
             this.fs,
             DirName.Manifest
         );
 
-        const shouldAddAnnotation = service.annotationUri && service.annotationUri.length > 0;
+        const shouldAddAnnotation = annotationId && annotationUri;
```

Comment:
I'd skip this line and put the expression directly in the if-block. Otherwise, people (like me) wonder where else in the code you use it.

Reply:
Fixed with: 9f6d864a51cb9db3fbfbc3f8e54db1a0822695b5